File: rixmsg/test/js/OtherMessage.js
import { MessageBase } from "../message_base.js";

export class OtherMessage extends MessageBase {
    constructor() {
        super();
        this.number = 0.0;
        this.flag = false;
    }

    size() {
        let size = 0;
        size += MessageBase._size_double();
        size += MessageBase._size_bool();
        return size;
    }

    hash() {
        return [BigInt(0x309c08666f8285b7), BigInt(0xe5ca3cc56cb0fbec)];
    }

    serialize(buffer) {
        buffer = MessageBase._serialize_double(this.number, buffer);
        buffer = MessageBase._serialize_bool(this.flag, buffer);
        return buffer;
    }

    deserialize(buffer, context) {
        this.number = MessageBase._deserialize_double(buffer, context);
        this.flag = MessageBase._deserialize_bool(buffer, context);
    }
}

File: rixmsg/test/python/OtherMessage.py
from rixmsg.message_base import MessageBase

class OtherMessage(MessageBase):
    def __init__(self):
        self.number = 0.0
        self.flag = False

    def size(self) -> int:
        size = 0
        size += MessageBase._size_double()
        size += MessageBase._size_bool()
        return size
    
    def serialize(self, buffer: bytearray) -> None:
        MessageBase._serialize_double(self.number, buffer)
        MessageBase._serialize_bool(self.flag, buffer)

    def deserialize(self, buffer: bytearray, context: dict) -> None:
        self.number = MessageBase._deserialize_double(buffer, context)
        self.flag = MessageBase._deserialize_bool(buffer, context)

    def hash(self) -> int:
        return [0x309c08666f8285b7, 0xe5ca3cc56cb0fbec]